radicle-cli/src/commands/patch.rs

```diff
@@ -215,12 +217,19 @@ pub fn run(options: Options, ctx: impl term::Context) -> anyhow::Result<()> {
             list::run(&repository, &profile, Some(workdir))?;
         }
         Operation::Show { ref patch_id } => {
-            show::run(&repository, &profile, &workdir, patch_id)?;
+            let patch_id = as_object_id(&git_repo, patch_id)?;
+            show::run(&repository, &profile, &workdir, &patch_id)?;
         }
         Operation::Update {
-            patch_id,
+            ref patch_id,
             ref message,
         } => {
+            let patch_id = if let Some(patch_id) = patch_id {
+                let patch_id = as_object_id(&git_repo, patch_id)?;
+                Some(patch_id)
+            } else {
+                None
+            };
```

Comment:
```suggestion
            let patch_id = patch_id.map(|id| as_object_id(&git_repo, id)).transpose()?;
```

Reply:
Ok, my brain blew up while I was trying to understand this!

Why do we need the transpose? in this case, the id is still an option that I need to unwrap anyway!

Reply:
Heh, so the idea is that `map` maps into the `Option`. So you end up with `Option<Result<ObjectId, Error>>`. Now, `transpose` pulls out the `Result` and pushes in the `Option` and you end up with `Result<Option<ObjectId>, Error>`. We then use `?` to bind `patch_id` to the `Option<ObjectId>`. In other words, it's the code you wrote but using `map` and `transpose` :)

I don't see an `unwrap` so I'm not sure what you mean :thinking:

Reply:
the id is an option `map(|id| repository_resolve_oid(id.unwrap())`, and I received the type mismatch :\

Reply:
I compiled again your solution

```rust
error[E0308]: mismatched types
   --> radicle-cli/src/commands/patch.rs:236:69
    |
236 |             let patch_id = patch_id.map(|id| repository.resolve_oid(patch_id)).transpose()?;
    |                                                         ----------- ^^^^^^^^ expected struct `ObjectId`, found enum `std::option::Option`
    |                                                         |
    |                                                         arguments to this function are incorrect
    |
    = note: expected struct `ObjectId`
                 found enum `std::option::Option<ObjectId>`
note: associated function defined here
   --> /home/vincent/Github/radicle/heartwood/radicle/src/storage/git.rs:419:12
    |
419 |     pub fn resolve_oid(&self, oid: cob::ObjectId) -> Result<cob::ObjectId, git2::Error> {
    |            ^^^^^^^^^^^

For more information about this error, try `rustc --explain E0308`.
```

Reply:
Ah I'm stupid! I found the error here `|id| repository.resolve_oid(patch_id)`